### DISCUSSION TITLE
setting up Toyota  integration

### DISCUSSION THREAD

**Discussion by dodampe-gamge** [DurgNomis-drol/ha_toyota#77] [Q&A] [answered]
Guys, 
I am new to HA but managed to installed HACS and Toyota Connected Services. how can configure it ? is it via yaml file ? 
any example showing how it should be to link specific toyota car ?

**Comment by DurgNomis-drol** [comment, marked as answer]
Integrations -> add integration -> Search `Toyota` and add it. You will now get asked to login with your Toyota account and that is it. The integration will create all available entities for all the cars you have set up in Toyota connected services. 😊

**Reply by DurgNomis-drol** [reply]
If you have not installed it through Hacs, you will need to do that first by going to Hacs -> integrations -> add integration and sear for Toyota and then add it. After that you will need to restart and proceed with above instructions 😊

**Reply by dodampe-gamge** [reply]
thanks.  it works after adding Toyota through integration.

**Reply by DurgNomis-drol** [reply]
Great to hear! Hope you like it!

**Comment by piasekps** [comment]
Hi
I tried to install and integrate your repo.
But I have some problems: 
- HACS doesn't find anything with Toyota...
- I tried to go manual install, I clone the repo and create symlink, but I restart whole HA and I can't see anywhere Toyota integration, neither on HACS > Integrations > add integration nor Configuration > integration > add integration.

Any idea what should I check?

**Reply by DurgNomis-drol** [reply]
What version of Home Assistant are you running?

**Reply by piasekps** [reply]
HA version 2021.12.9

**Reply by DurgNomis-drol** [reply]
That's strange. 

Are you coping the files or creating a symlink? The files need to be placed in `custom_components/toyota`

It seems that there are something wrong with your install if you can't find it on HACS. Are you running the lastest version of HACS?

**Reply by piasekps** [reply]
Perfect updating HACS to the newest version help. I was not aware new version was not installed. Thx

**Reply by piasekps** [reply]
I have few questions, is that normally: 

-  I do not have all sensors visible on the HA? (f.e. sensor.corrola)
- Sensors after a few min shows status 'unavailable'
- is there a chance to get a specific warning status I know we have binary_sensor.corolla_over_all_status but to get single status value?
- is there a chance to track f.e. low wash water warning?
- is there a way to get a car position while driving?
Is there a place to propose some features?
f.e. sensor which says if a car is open or locked (should check all doors)

**Reply by DurgNomis-drol** [reply]
1. Sensors available depends on what car model you have and production year. There is no way of telling what is support other then seeing what is created when adding the integration.
2. They shouldn't do that, that is probably a bug. The library this integration depends on is being updated which should fix a lot of things. But please open a issue so we can track it.
3. All sensors have an attribute that tells you the status of that specific thing (Window, door, etc...)
4. No unfortunately not at this moment. Hopefully in the future. If my Toyota Android App supports it, it is possible to add support for it.
5. Not as far as I know.
6. Door and Window sensors will tell you if they are locked/opened or not. You can create a template sensor for checking all doors/windows or an automation that sends you a notification if a doors open/unlocked and you are not near it.
